{
    "source": "BDSA",
    "name": "BDSA-2021-1701",
    "title": "upx Vulnerable to Denial-of-Service (DoS) via Assertion Failure in 'MemBuffer::alloc()' Function",
    "description": "upx contains a denial-of-service (DoS) vulnerability. An attacker can take advantage of this by using a crafted file to force an assertion failure and crash the application.",
    "technicalDescription": "This vulnerability exists in the `MemBuffer::alloc()` function of file `src/p_mach.cpp`. The `MemBuffer` is attempted to be allocated with 0 bytes, failing an assertion defined in `mem.cpp`.",
    "publishedDate": "2021-06-07T16:10:23.820Z",
    "updatedDate": "2022-04-07T11:11:46.296Z",
    "disclosureDate": "2021-04-07T00:00:00.000Z",
    "exploitPublishDate": "2021-04-07T00:00:00.000Z",
    "solution": "Fixed in the **devel** branch by [this](https://github.com/upx/upx/commit/28e761cd42211dfe0124b7a29b2f74730f453e46) commit, which is not yet included in any commit.",
    "severity": "MEDIUM",
    "cvss2": {
        "baseScore": 5.0,
        "impactSubscore": 2.9,
        "exploitabilitySubscore": 10.0,
        "severity": "LOW",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "PROOF_OF_CONCEPT",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 3.9
        },
        "vector": "(AV:N/AC:L/Au:N/C:N/I:N/A:P/E:POC/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 5.3,
        "impactSubscore": 1.4,
        "exploitabilitySubscore": 3.9,
        "severity": "MEDIUM",
        "attackVector": "NETWORK",
        "attackComplexity": "LOW",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "LOW",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "PROOF_OF_CONCEPT",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 4.8
        },
        "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L/E:P/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": true,
    "parentAdvisory": false,
    "credit": "chibataiki",
    "workaround": "",
    "vendorFixDate": "2021-04-10T00:00:00.000Z",
    "bdsaTags": [],
    "overallScore": 4.8,
    "_meta": {
        "allow": [
            "GET"
        ],
        "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2021-1701",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro2.app.blackduck.com/api/cwes/CWE-400"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/CVE-2021-30501",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2021-1701/ranges"
            },
            {
                "rel": "reference",
                "href": "https://github.com/upx/upx/commit/28e761cd42211dfe0124b7a29b2f74730f453e46",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/upx/upx/issues/486",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/upx/upx/issues/486",
                "label": "POC"
            },
            {
                "rel": "default-remediation-status",
                "href": "https://trendmicro2.app.blackduck.com/api/vulnerabilities/BDSA-2021-1701/default-remediation-status"
            }
        ]
    }
}